sale_stock_picking_note/models/sale_stock.py

```diff
@@ -0,0 +1,23 @@
+# Copyright 2018 Carlos Dauden - Tecnativa <carlos.dauden@tecnativa.com>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    picking_note = fields.Text(
+        string="Picking Note",
+    )
+
+
+class StockMove(models.Model):
+    _inherit = 'stock.move'
+
+    def _get_new_picking_values(self):
+        vals = super()._get_new_picking_values()
+        sale_note = self.sale_line_id.order_id.picking_note
+        if sale_note:
+            vals['note'] = sale_note
```

Comment:
@chienandalu IMO here use `vals.get('note')` and check in If condition `if sale_note and vals.get('note'):`

Reply:
I have put a better formula IMO